File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/itemCatService.js
//服务层
app.service('itemCatService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../itemCat/findAll.php');
	};
	//分页 
	this.findByPagination=function(page,rows){
		return $http.get('../itemCat/findByPagination.php?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findById=function(id){
		return $http.get('../itemCat/findById.php?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../itemCat/add.php',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../itemCat/update.php',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../itemCat/delete.php?ids='+ids);
	};
	//搜索
	this.findWithConditionsByPagination=function(page,rows,searchEntity){
		return $http.post('../itemCat/findWithConditionsByPagination.php?page='+page+"&rows="+rows, searchEntity);
	};
    //根据上级ID查询下级列表
    this.findByParentId=function(parentId){
        return $http.get('../itemCat/findByParentId.php?parentId='+parentId);
    }


});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/cn/porkchop/pinyougou/sellergoods/service/ItemCatService.java
package cn.porkchop.pinyougou.sellergoods.service;

import cn.porkchop.pinyougou.pojo.PageResult;
import cn.porkchop.pinyougou.pojo.TbItemCat;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface ItemCatService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbItemCat> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    PageResult findByPagination(int pageNum, int pageSize);


    /**
     * 增加
     */
    void add(TbItemCat itemCat);


    /**
     * 修改
     */
    void update(TbItemCat itemCat);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    TbItemCat findById(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum
     *         当前页 码
     * @param pageSize
     *         每页记录数
     * @return
     */
    PageResult findWithConditionsByPagination(TbItemCat itemCat, int pageNum, int pageSize);

    /**
     * 根据上级id返回列表
     *
     * @date 2018/6/25 14:36
     * @author porkchop
     */
    List<TbItemCat> findByParentId(Long parentId);
}
